File: app/src/main/java/com/example/app/restful/_WebService.java
package com.example.app.restful;

import com.example.app.model.JobPosting;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/*
https://jobs.github.com/api
*/
public interface _WebService {

    /*
        https://jobs.github.com/positions.json?description=python&full_time=true&location=sf&page=1
     */
    @GET("positions.json?full_time=true&location=sf&page=1")
    Call<List<JobPosting>> getList(@Query("description") String description);

}

File: app/src/main/java/com/example/app/restful/_WebService.kt
package com.example.app.restful

import com.example.app.model.JobPosting

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

/*
https://jobs.github.com/api
*/
interface _WebService {

    /*
        https://jobs.github.com/positions.json?description=python&full_time=true&location=sf&page=1
     */
    @GET("positions.json?full_time=true&location=sf&page=1")
    fun getList(@Query("description") description: String): Call<List<JobPosting>>

    @GET("/positions/{id}.json")
    fun getId(@Path("id") id: String): Call<JobPosting>
}